dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -493,6 +493,15 @@ private Map<String, List<String>> getDimensionItems( EventQueryParams params,
 
             if ( item.hasOptionSet() )
             {
+                // if itemOptions.get( item.getItem().getUid() ) returns null,
```

Comment:
I think you could make this comment a bit clearer. I would split it into two. The first part is below, and the second part is at the end of the code review.

```
// The call itemOptions.get( item.getItem().getUid() ) can return null.
// This should be ok, the query item can't have both legends and options.
```